nifi-nar-bundles/nifi-html-bundle/nifi-html-processors/src/test/java/org/apache/nifi/TestGetHTMLElement.java

```diff
@@ -0,0 +1,319 @@
+
+        ProcessSession session = testRunner.getProcessSessionFactory().createSession();
+        FlowFile ff = writeContentToNewFlowFile(HTML.getBytes(), session);
+
+        testRunner.enqueue(ff);
+        testRunner.run();
+
+        testRunner.assertTransferCount(GetHTMLElement.REL_SUCCESS, 3);
+        testRunner.assertTransferCount(GetHTMLElement.REL_FAILURE, 0);
+        testRunner.assertTransferCount(GetHTMLElement.REL_ORIGINAL, 1);
+        testRunner.assertTransferCount(GetHTMLElement.REL_NOT_FOUND, 0);
+    }
+
+    @Test
+    public void testElementFoundWriteToAttribute() throws Exception {
+        testRunner.setProperty(GetHTMLElement.CSS_SELECTOR, "#" + ATL_ID);
+        testRunner.setProperty(GetHTMLElement.DESTINATION, GetHTMLElement.DESTINATION_ATTRIBUTE);
+        testRunner.setProperty(GetHTMLElement.OUTPUT_TYPE, GetHTMLElement.ELEMENT_ATTRIBUTE);
+        testRunner.setProperty(GetHTMLElement.ATTRIBUTE_KEY, "href");
+
+        ProcessSession session = testRunner.getProcessSessionFactory().createSession();
+        FlowFile ff = writeContentToNewFlowFile(HTML.getBytes(), session);
+
+        testRunner.enqueue(ff);
+        testRunner.run();
+
+        testRunner.assertTransferCount(GetHTMLElement.REL_SUCCESS, 1);
+        testRunner.assertTransferCount(GetHTMLElement.REL_FAILURE, 0);
+        testRunner.assertTransferCount(GetHTMLElement.REL_ORIGINAL, 1);
+        testRunner.assertTransferCount(GetHTMLElement.REL_NOT_FOUND, 0);
+
+        List<MockFlowFile> ffs = testRunner.getFlowFilesForRelationship(GetHTMLElement.REL_SUCCESS);
+        assertTrue(ffs.size() == 1);
+        MockFlowFile fff = ffs.get(0);
+        String atValue = fff.getAttribute(GetHTMLElement.HTML_ELEMENT_ATTRIBUTE_NAME);
+        assertTrue(StringUtils.equals(ATL_WEATHER_LINK, atValue));
+    }
+
+    @Test
+    public void testElementFoundWriteToContent() throws Exception {
+        testRunner.setProperty(GetHTMLElement.CSS_SELECTOR, "#" + ATL_ID);
+        testRunner.setProperty(GetHTMLElement.DESTINATION, GetHTMLElement.DESTINATION_CONTENT);
+        testRunner.setProperty(GetHTMLElement.OUTPUT_TYPE, GetHTMLElement.ELEMENT_ATTRIBUTE);
+        testRunner.setProperty(GetHTMLElement.ATTRIBUTE_KEY, "href");
+
+        ProcessSession session = testRunner.getProcessSessionFactory().createSession();
+        FlowFile ff = writeContentToNewFlowFile(HTML.getBytes(), session);
+
+        testRunner.enqueue(ff);
+        testRunner.run();
+
+        testRunner.assertTransferCount(GetHTMLElement.REL_SUCCESS, 1);
+        testRunner.assertTransferCount(GetHTMLElement.REL_FAILURE, 0);
+        testRunner.assertTransferCount(GetHTMLElement.REL_ORIGINAL, 1);
+        testRunner.assertTransferCount(GetHTMLElement.REL_NOT_FOUND, 0);
+
+        List<MockFlowFile> ffs = testRunner.getFlowFilesForRelationship(GetHTMLElement.REL_SUCCESS);
+        assertTrue(ffs.size() == 1);
+        String data = new String(testRunner.getContentAsByteArray(ffs.get(0)));
```

Comment:
Recommend we go with:
ffs.get(0).assertContentEquals(ATL_WEATHER_LINK);

This gives us a nice output if the unit test were to fail, and would indicate what we expected the value to be vs. what it actually was.